Added docs section for using secrets manager with the extension (#222)
elastic/apm-aws-lambda: docs/monitoring-aws-lambda.asciidoc

[[aws-lambda-secrets-manager]]
== Using AWS Secrets Manager to manage APM authentication keys
When using the config options <<aws-lambda-config-authentication-keys>> for authentication of the APM Lambda Extension with Elastic APM the corresponding keys are specified in plain text in the environment variables of your Lambda function. If you prefer to securely store the authentication keys, you can use the AWS Secrets Manager and let the APM Lambda Extension retrieve the actual keys from the AWS Secrets Manager. Follow the instructions below to setup the AWS Secrets Manager with the APM Lambda Extension.

[[aws-lambda-secrets-manager-create-secret]]
=== Step 1: Create a secret in the AWS Secrets Manager.
https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html[Create a secret in the AWS Secrets Manager] for the {apm-guide-ref}/secret-token.html[APM Secret Token] or the {apm-guide-ref}/api-key.html[APM API key], depending on which one you prefer to use. Make sure to create the secret as a *Plaintext* typed secret and ensure it is created *in the same AWS region* as your target Lambda function that will use the secret. 

> We recommend to use the AWS managed encryption key `aws/secretsmanager`. However, you can optionally create and select a custom KMS key for encryption. Note that with a custom encryption key you will need additional key permissions on your Lambda function (see <<aws-lambda-secrets-manager-permissions, Step 2>>).

Reviewer: ```suggestion
We recommend using the AWS-managed encryption key `aws/secretsmanager`. However, you can optionally create and select a custom KMS key for encryption. Note that with a custom encryption key, you will need additional key permissions on your Lambda function (see <<aws-lambda-secrets-manager-permissions, Step 2>>).
```

include::./secrets-manager/create-secret-widget.asciidoc[]

Remember your chosen secret name. You will use the secret name as the value for the config options <<aws-lambda-config-secrets-manager-options>> when configuring your APM Lambda Extension.

[[aws-lambda-secrets-manager-permissions]]
=== Step 2: Add permissions to your AWS Lambda function

> For your Lambda function to be able to retrieve the authentication key from the AWS Secrets Manager you need to provide the following permissions to your Lambda function.

Reviewer: ```suggestion
For your Lambda function to be able to retrieve the authentication key from the AWS Secrets Manager, you need to provide the following permissions to your Lambda function.
```

include::./secrets-manager/add-permissions-widget.asciidoc[]

=== Step 3: Configure the APM Lambda extension
Finally, you will need to configure the APM Lambda extension to make it use the secret from the Secrets Manager _instead of_ the value provided through <<aws-lambda-config-authentication-keys>>.

Provide the name of the secret you created in <<aws-lambda-secrets-manager-create-secret, Step 1>> as the value for the <<aws-lambda-config-secrets-manager-options>> config option, respectively, depending on whether you want to use the {apm-guide-ref}/secret-token.html[APM Secret Token] or the {apm-guide-ref}/api-key.html[APM API key]. 

> The language specific instructions describe how to set environment variables for the purpose of configuring AWS Lambda for Elastic APM:

Reviewer: "for the purpose of" feels redundant here

```suggestion
The language-specific instructions describe how to set environment variables for configuring AWS Lambda for Elastic APM:
```
